repronim/interface/tests/test_install.py

```diff
@@ -9,13 +9,95 @@
+from repronim.tests.utils import with_tree
+from repronim.cmd import Runner
+
+
+# Sample output from Reprozip.
+REPROZIP_OUTPUT = """
+runs:
+# Run 0
+- architecture: x86_64
+  date: 2016-02-22T22:19:01.735754
+  argv: [echo, $MATH_EXPRESSION, '|', bc]
+  binary: /usr/bin/bc
+  distribution: [Ubuntu, '12.04']
+  environ: {TERM: xterm, MATH_EXPRESSION: '2+2'}
+  exitcode: 0
+  gid: 1003
+  hostname: nitrcce
+  id: run0
+  system: [Linux, 3.2.0-98-virtual]
+  uid: 1002
+  workingdir: /home/rbuccigrossi/simple_workflow
+packages:
+  - name: "base-files"
+    version: "6.5ubuntu6.8"
+    size: 429056
+    packfiles: true
+    files:
+      # Total files used: 103.0 bytes
+      # Installed package size: 419.00 KB
+      - "/etc/debian_version" # 11.0 bytes
+      - "/etc/host.conf" # 92.0 bytes
+  - name: "bc"
+    version: "1.06.95-2ubuntu1"
+    size: 1449984
+    packfiles: true
+    files:
+      # Total files used: 936.64 KB
+      # Installed package size: 1.38 MB
+      - "/bin/bash" # 936.64 KB
+"""
+
+@with_tree(tree={'sample.yml': REPROZIP_OUTPUT})
+def test_install_packages_localhost(path):
+    """Test installing 2 packages on the localhost.
     """
-    Install two packages locally: base-files and bash
+    testfile = pathjoin(path, 'sample.yml')
+    with patch.object(Runner, 'run', return_value='installed package') as mocked_call, \
+        swallow_logs(new_level=logging.DEBUG) as log:
+
+        main(['install', '--spec', testfile, '--platform', 'localhost'])
+
+        calls = [call(['apt-get', 'install', '-y', package], shell=True) for package in ('base-files', 'bc')]
+        mocked_call.assert_has_calls(calls)
+
+        assert_in("Installing package: base-files", log.lines)
+        assert_in("Installing package: bc", log.lines)
+        assert_in("installed package", log.lines)
+
+@with_tree(tree={'sample.yml': REPROZIP_OUTPUT})
+def test_install_packages_dockerengine(path):
+    """Test installing 2 packages into a Docker container.
     """
-    testfile = dirname(abspath(__file__)) + os.path.sep + 'sample_reprozip_output_small.yml'
-    main(['install', '--spec', testfile, '--platform', 'localhost'])
+    testfile = pathjoin(path, 'sample.yml')
+    with patch('docker.Client') as MockClient, swallow_logs(new_level=logging.DEBUG) as log:
+
+        # Set up return values for mocked docker.Client methods.
+        client = MockClient.return_value
+        client.build.return_value = ['{"stream": "Successfully built 9a754690460d\\n"}']
+        client.create_container.return_value = {u'Id': u'd4cb4ee', u'Warnings': None}
+        client.start.return_value = None
+        client.logs.return_value = 'container standard output'
+
+        args = ['install',
+                    '--spec', testfile,
+                    '--platform', 'dockerengine',
+                    '--host', 'tcp://127.0.0.1:2375',
+                    '--image', 'repronim_test']
+        main(args)
 
+        assert client.build.called
```

Comment:
Let me convert to nose. I believe I got on the py.test track simpley because my PyCharm IDE started using it by default. I am having problems of tests running fine on my dev env and then getting failed by Travis. Let me make the change on my side and resubmit my pull request.